src/CloudNative.CloudEvents/CloudEvent.cs

```diff
@@ -390,10 +390,11 @@ public CloudEvent Validate()
         /// <summary>
         /// Validates that this CloudEvent is valid in the same way as <see cref="IsValid"/>,
         /// but throwing an <see cref="ArgumentException"/> using the given parameter name
-        /// if the event is invalid. This is typically used within protocol bindings as the last
-        /// step in conversion.
+        /// if the event is invalid. This is typically used within protocol bindings or event formatters
+        /// as the last step in decoding an event, or as the first step when encoding an event.
         /// </summary>
-        /// <param name="paramName">The parameter name to use in the exception if the event is invalid.</param>
+        /// <param name="paramName">The parameter name to use in the exception if the event is invalid.
+        /// May be null.</param>
         /// <exception cref="ArgumentException">The event is invalid.</exception>
         /// <returns>A reference to the same object, for simplicity of method chaining.</returns>
         public CloudEvent ValidateForConversion(string paramName)
```

Comment:
What if this was just called `Validate`?

Reply:
There's already a more "normal" `Validate` method which throws `InvalidOperationException`.

I think moving `ValidateForConversion` to an extension method in a different namespace will be the best option. Will do that in a later PR.